src/ui/marker.js

```diff
@@ -508,6 +523,10 @@ export default class Marker extends Evented {
         }
 
         DOM.setTransform(this._element, `${anchorTranslate[this._anchor]} translate(${this._pos.x}px, ${this._pos.y}px) ${pitch} ${rotation}`);
+
+        if (this._map.getTerrain() || this._map.getFog() && !this._fadeTimer) {
```

Comment:
Possible to add parentheses to clarify the intent? I think it's this:

```suggestion
        if ((this._map.getTerrain() || this._map.getFog()) && !this._fadeTimer) {
```